### PR TITLE
Add AWS classses

### PR DESCRIPTION
**Pull request by mbj**
### Acceptance

* [x] Green CI

### CODE REVIEW

**Comment by mbj** [comment]
@epicallan This is ready for review.

File: amazonka/src/MRIO/Amazonka.hs [hunk 1]
```diff
@@ -1,28 +1,73 @@
...
 
-import qualified Network.AWS                   as AWS
+import qualified Network.AWS as AWS
+
+class HasAWSEnv env where
+  awsEnv :: env -> AWS.Env
+
+type ResourceMap = IORef ReleaseMap
+
+class HasResourceMap env where
```
Comment (mbj) [inline amazonka/src/MRIO/Amazonka.hs:42]: @epicallan In the future we may want to split this into an `mrio-resourcet` package.

File: amazonka/src/MRIO/Amazonka.hs [hunk 2]
```diff
@@ -1,28 +1,73 @@
...
+  resourceMap :: env -> ResourceMap
 
-class HasAWS env where
-  runAWS :: AWS a -> RIO env a
+instance HasResourceMap env => MonadResource (RIO env) where
```
Comment (mbj) [inline amazonka/src/MRIO/Amazonka.hs:45]: This would be part of the `mrio-resourcet` package.

Comment (epicallan) [reply amazonka/src/MRIO/Amazonka.hs:45]: Sounds good.

**Pull request merged**